src/algebra/direct_sum.lean

```diff
@@ -104,12 +106,31 @@ def to_add_monoid : (⨁ i, β i) →+ γ :=
 @[simp] lemma to_add_monoid_of (i) (x : β i) : to_add_monoid φ (of β i x) = φ i x :=
 dfinsupp.lift_add_hom_apply_single φ i x
 
-variables (ψ : (⨁ i, β i) →+ γ)
-
 theorem to_add_monoid.unique (f : ⨁ i, β i) :
   ψ f = to_add_monoid (λ i, ψ.comp (of β i)) f :=
 by {congr, ext, simp [to_add_monoid, of]}
 
+end to_add_monoid
+
+section from_add_monoid
+
+/-- `from_add_monoid φ` is the natural homomorphism from `γ` to `⨁ i, β i`
+induced by a family `φ` of homomorphisms `γ → β i`.
+
+Note that this is not an isomorphism. Not every homomorphism `γ →+ ⨁ i, β i` arises in this way. -/
+def from_add_monoid : (⨁ i, γ →+ β i) →+ (γ →+ ⨁ i, β i) :=
+to_add_monoid $ λ i, add_monoid_hom.comp_hom (of β i)
```

Comment:
Does it make sense to putt this in `dfinsupp` (and `finsupp`, as `(i →₀ γ →+ β) →+ (γ →+ i →₀ β)`) too?

Reply:
cc @urkud who added `finsupp.lift_add_hom` in the first place.

Reply:
@eric-wieser I don't know... what will be the benefit of `direct_sum` over `dfinsupp`? Should we just remove `direct_sum` and make it optional notation for `dfinsupp`? Where exactly should the API boundary lie? I've never really used `dfinsupp`.

Reply:
I already don't know what the benefit of `direct_sum` over `dfinsupp` is. My guess is that the best comparison is `monoid_algebra` vs `finsupp`, where the former gets convolutive multiplication while the latter gets pointwise multiplication.

Of course, this is currently irrelevant for `direct_sum` and `dfinsupp` because neither gets any multiplication!

Reply:
hmm, then I would prefer to just do this for `direct_sum` for now. I tried copying the def to `dfinsupp`, but it didn't *just work*. And for me it's clear that this is something that we want for direct sums.
If we decide to merge the two concepts, then this PR is certainly not the place for that.

Reply:
Fair enough